packages/data-provider/src/services/submittedDataService.ts

```diff
@@ -315,11 +443,265 @@ const service = (dependencies: BaseDependencies) => {
+	 * @param systemId - The unique identifier for the system associated with the submitted data.
+	 * @param filterOptions - An object containing options for data representation.
+	 * @param filterOptions.view - The desired view type for the data representation, such as 'list' or 'compound'.
+	 * @returns A promise that resolves to an object containing:
+	 * - `data`: The fetched `SubmittedDataResponse`, or `undefined` if no data is found.
+	 * - `metadata`: An object containing metadata about the fetched data, including an optional `errorMessage` property.
+	 */
+	const getSubmittedDataBySystemId = async (
+		categoryId: number,
+		systemId: string,
+		filterOptions: { view: ViewType },
+	): Promise<{
+		data: SubmittedDataResponse | undefined;
+		metadata: { errorMessage?: string };
+	}> => {
+		// get SubmittedData by SystemId
+		const foundRecord = await submittedDataRepo.getSubmittedDataBySystemId(systemId);
+		logger.info(LOG_MODULE, `Found Submitted Data with system ID '${systemId}'`);
+
+		if (!foundRecord) {
+			return { data: undefined, metadata: { errorMessage: `No Submitted data found with systemId '${systemId}'` } };
+		}
+
+		if (foundRecord.dictionaryCategoryId !== categoryId) {
+			return {
+				data: undefined,
+				metadata: { errorMessage: `Invalid Category ID '${categoryId}' for system ID '${systemId}'` },
+			};
+		}
+
+		let dataValue: DataRecordNested = foundRecord.data;
+
+		if (filterOptions.view === VIEW_TYPE.Values.compound) {
+			const { getCategoryById } = categoryRepository(dependencies);
+
+			const category = await getCategoryById(foundRecord.dictionaryCategoryId);
+
+			if (!category?.activeDictionary) {
+				return { data: undefined, metadata: { errorMessage: `Invalid Category ID` } };
+			}
+
+			// Retrieve compound records only if the record matches the type of the default Centric Entity
+			// or if no default Centric Entity is defined.
+			if (
+				!category.defaultCentricEntity ||
+				(category.defaultCentricEntity && category.defaultCentricEntity === foundRecord.entityName)
+			) {
+				// get dictionary hierarchy structure
+				const hierarchyStructureDesc = generateHierarchy(category.activeDictionary.dictionary, ORDER_TYPE.Values.desc);
+				const hierarchyStructureAsc = generateHierarchy(category.activeDictionary.dictionary, ORDER_TYPE.Values.asc);
+
+				const childNodes = await traverseChildNodes({
+					data: foundRecord.data,
+					entityName: foundRecord.entityName,
+					organization: foundRecord.organization,
+					schemaCentric: foundRecord.entityName,
+					treeNode: hierarchyStructureDesc,
+				});
+
+				const parentNode = await traverseParentNodes({
+					data: foundRecord.data,
+					entityName: foundRecord.entityName,
+					organization: foundRecord.organization,
+					schemaCentric: foundRecord.entityName,
+					treeNode: hierarchyStructureAsc,
+				});
+
+				dataValue = { ...dataValue, ...childNodes, ...parentNode };
+			}
+		}
+
+		return {
+			data: {
+				data: dataValue,
```

Comment:
`data.data`? is it possible to then have data.data.data? etc.

Reply:
Probably not good at naming variables 😄. `result.data` works fine. Thanks for the suggestion!

Reply:
This comment will be handled in PR[#105](https://github.com/overture-stack/lyric/pull/105)